components/automate-gateway/handler/applications.go

```diff
@@ -24,18 +26,24 @@ func NewApplicationsHandler(applicationsClient applications.ApplicationsServiceC
 // GetServiceGroupsHealthCounts returns the health counts from all service groups
 func (a *Applications) GetServiceGroupsHealthCounts(
 	ctx context.Context,
-	in *applications.ServiceGroupsHealthCountsReq) (*applications.HealthCounts, error) {
+	request *applications.ServiceGroupsHealthCountsReq) (*applications.HealthCounts, error) {
 
-	inDomain := &applications.ServiceGroupsHealthCountsReq{}
-	out := &applications.HealthCounts{}
-	f := func() (proto.Message, error) {
-		return a.client.GetServiceGroupsHealthCounts(ctx, inDomain)
-	}
-	err := protobuf.CallDomainService(in, inDomain, f, out)
+	log.WithFields(log.Fields{
+		"request": request.String(),
+		"func":    nameOfFunc(),
+	}).Debug("rpc call")
+
+	response, err := a.client.GetServiceGroupsHealthCounts(ctx, &applications.ServiceGroupsHealthCountsReq{})
 	if err != nil {
 		return nil, err
 	}
-	return out, nil
+	return &applications.HealthCounts{
+		Total:    response.Total,
+		Ok:       response.Ok,
+		Warning:  response.Warning,
+		Critical: response.Critical,
+		Unknown:  response.Unknown,
+	}, nil
```

Comment:
I believe under certain circumstances, the compiler will do this for us: you could try
```go
return (*applications.HealthCounts)(response), nil
```
It's only allowed if the field types match, I believe. (I.e. if the compiler lets you do it, it's fine 😄 )

Reply:
awesome! yes, the field types match, if fact, they are the same so I just found a better way to do this. 💯